test/hyphenation_eval/resources/generate_hyphenation_test_data.py

```
@@ -16,28 +17,114 @@
+    r"\*{3}\s*END OF (?:THIS|THE) PROJECT GUTENBERG EBOOK.*?\*{3}",
+    flags=re.IGNORECASE,
+)
+LEGACY_GUTENBERG_END_RE = re.compile(
+    r"\bEnd of (?:the )?Project Gutenberg", flags=re.IGNORECASE
+)
+
+
+class EpubTextExtractor(HTMLParser):
+    """Extract visible book text while excluding EPUB navigation and boilerplate."""
+
+    SKIPPED_TAGS = {"head", "nav", "script", "style"}
```

Ruff (0.15.20)

[warning] Mutable default value for class attribute

(RUF012)